### PR TITLE
[Refactor] use three-way comparison

### PR DESCRIPTION
**Pull request by ucasfl**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/util/uid_util.h [hunk 1]
```diff
@@ -97,22 +97,11 @@ struct UniqueId {
...
 
     // std::map std::set needs this operator
-    bool operator<(const UniqueId& right) const {
-        if (hi != right.hi) {
-            return hi < right.hi;
-        } else {
-            return lo < right.lo;
-        }
-    }
+    auto operator<=>(const UniqueId& right) const = default;
```
Comment (kevincai) [inline be/src/util/uid_util.h:100]: suggest keep this function, the default behavior depends on the memory layout of the member variables.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/util/uid_util.h (hunk 1 above)
Comment (ucasfl) [reply be/src/util/uid_util.h:100]: But the memory layout of the class members is determined?

Comment (kevincai) [reply be/src/util/uid_util.h:100]: if new field is added, or code refactored, it is easy to cause issues without even noticed. probably UT shall cover the comparison behavior.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Pull request closed without merging**